sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -666,11 +692,34 @@ const StartHTLCAssetLockListener = (
     assetId: string,
     recipientECert: string,
     lockerECert: string,
-    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: string) => any,
+    lockCallback: (c: Contract, d: string, t: string, i: string, r: string, l: string, v: Hash, timeout: number) => any,
 ): void => {
     StartHTLCEventListener(contract, 'LockAsset', contractId, assetType, assetId, -1, recipientECert, lockerECert, lockCallback);
 }
 
+//TODO: Timed counterpart for timed AssetLockListener for reversion
+const assetLockCallbackTimed = (
+    contract: Contract, 
+    contractID: string, 
+    assetType: string, 
+    assetID: string, 
+    recipientECert: string, 
+    lockerECert: string, 
+    hashMechanism: Hash,
+    expiryTime: number
+): void => {
+    //Compare expiryTimeSec with currentTimeSec, which is number of milliseconds since epoch (L174)
+    const currTimeSecs = Math.floor(Date.now()/1000);  
+    if(expiryTime > currTimeSecs) {
+        claimAssetInHTLC(contract, assetType, assetID, recipientECert, hashMechanism);
+    }
+
+    //If we want to wait until timeout, we need to "sleep till"
+    else if(expiryTime <= currTimeSecs) {
+        reclaimAssetInHTLC(contract, assetType, assetID, recipientECert);
```

Comment:
You can't just call this function here directly. You need to nest it within a callback within a `SetTimeout`.